File: android/src/main/java/com/red5pro/reactnative/module/R5StreamItem.java
package com.red5pro.reactnative.module;

import com.red5pro.reactnative.stream.R5StreamInstance;
import com.red5pro.streaming.config.R5Configuration;

public class R5StreamItem {

	private R5Configuration configuration;
	private R5StreamInstance instance;

	public R5StreamItem (R5Configuration configuration) {
		this.configuration = configuration;
	}

	public R5Configuration getConfiguration() {
		return configuration;
	}

	public R5StreamInstance getInstance() {
		return instance;
	}

	public void setInstance(R5StreamInstance instance) {
		this.instance = instance;
	}

}

File: ios/R5VideoView/R5VideoView/R5StreamItem.m
//
//  R5StreamItem.m
//  R5VideoView
//
//  Created by Todd Anderson on 03/12/2018.
//  Copyright © 2018 Red5Pro. All rights reserved.
//

#import "R5StreamItem.h"
#import "R5StreamInstance.h"

@interface R5StreamItem() {
    
    R5Configuration *_configuration;
    NSObject<R5StreamInstance> *_streamInstance;
    
}
@end


@implementation R5StreamItem

- (void)clear {
    _configuration = nil;
    _streamInstance = nil;
}

- (id)initWithConfiguration:(R5Configuration *)configuration {
    
    if (self = [super init]) {
        _configuration = configuration;
    }
    return self;
    
}

- (R5Configuration *)getConfiguration {
    return _configuration;
}

- (NSObject<R5StreamInstance> *)getStreamInstance {
    return _streamInstance;
}

- (void)setStreamInstance:(NSObject<R5StreamInstance> *)streamInstance {
    _streamInstance = streamInstance;
}

@end
